runtime/src/bank.rs

```diff
@@ -626,12 +627,73 @@ impl Bank {
             &sysvar::rewards::create_account(1, validator_point_value, storage_point_value),
         );
 
+        let validator_rewards = self.pay_validator_rewards(validator_point_value);
+
         self.capitalization.fetch_add(
-            (validator_rewards + storage_rewards) as u64,
+            validator_rewards + storage_rewards as u64,
             Ordering::Relaxed,
         );
     }
 
+    /// iterate over all stakes, redeem vote credits for each stake we can
+    ///   successfully load and parse, return total payout
+    fn pay_validator_rewards(&self, validator_point_value: f64) -> u64 {
+        let stake_history = self.stakes.read().unwrap().history().clone();
+        self.stake_delegations()
+            .iter()
+            .map(
+                |(
+                    stake_pubkey,
+                    Delegation {
+                        voter_pubkey: vote_pubkey,
+                        ..
+                    },
+                )| {
+                    match (
+                        self.get_account(&stake_pubkey),
```

Comment:
Since you return zero in all the edge cases, how about moving all this code into a function that returns an Option, so that you can get rid of all the nested blocks:

```rust
let mut stake_state = self.get_account(&stake_pubkey)?.state().ok()?;
let mut vote_state = self.get_account(&vote_pubkey)?.state.ok()?;
```

Reply:
when I moved the meat of the function into a separate function, I ended up with a 7 or 8 parameter function...

Reply:
Sounds like a strong indictor something needs to be done here. Maybe multiple functions!

Reply:
I considered collecting what needed to be stored to flatten out the code.  I also tried a single, 4-parameter match but ran afoul of the borrow checker.